src/scenes/AcceleratorRouter/ParticipantProjects/EditView/ProjectProfileEdit.tsx

```diff
@@ -235,16 +324,46 @@ const ProjectProfileEdit = () => {
     return false;
   }, [tfContact, dispatch, project?.organizationId]);
 
+  const saveInternalUsers = useCallback(() => {
+    if (listInternalUsersRequest?.data?.users) {
+      const updateRequest = dispatch(
+        requestProjectInternalUsersUpdate({
+          projectId,
+          usersToAssign: (internalUsers || [])
+            .filter((user) => !!user.userId)
+            .map((user) => ({
+              role: user.role,
+              roleName: user.roleName,
+              userId: user.userId as number,
+            })),
+          usersToRemove: listInternalUsersRequest?.data?.users || [],
```

Comment:
I think this may cause a race condition if a user is not being added or removed. This is saying to remove all users and then reassign some of them, right? And in the thunk we're calling all the removal endpoints and then the assign endpoints, so it seems like it's possible that a user could be re-assigned and then removed if the BE processes them in varying speeds. Am I thinking about that right?

Reply:
The requests are initiated in the proper order in the FE, if it's possible for the BE to complete the work out-of-order perhaps we need to re-think the API for this to require fewer requests?

Reply:
In the case that only the role is changed for an existing user, the FE would still need to remove and re-add the modified user. Let's go with option 2.

Reply:
Ok. I'm deep in survival rate stuff so it'll be a bit before I can switch back to that. Will let you know once it's merged

Reply:
I'm not sure I follow your point about requests being initiated without waiting. From what I see, the FE waits for all remove requests to complete before starting the assign requests.

Reply:
Ah you're right, sorry. Rather than having you rewrite the API, I can separate these so the remove requests complete before starting the assign requests.

Reply:
@ice1080 I went ahead and cleaned this function up because it was bugging me and  it only took a couple of minutes. I'll plan on updating the FE once the API changes are ready. No rush on my end at all, I have other tasks I can focus on.